backend/src/controllers/clientCourses/manageAssessment.ts

```diff
@@ -0,0 +1,45 @@
+import { NextFunction, Request, Response } from 'express';
+
+import { AssessmentAction } from 'enums/common';
+import CourseStatus from 'enums/coursesEnums';
+import BadRequestError from 'classes/errors/clientErrors/BadRequestError';
+import { getStatusProvider, updateClientCourseField } from 'db/providers/clientCourseProvider';
+import { COURSE_FILEDS } from 'config/constants';
+
+const manageAssessment = async (
+  req: Request<{ id: string }, unknown, { action: AssessmentAction }>,
+  res: Response<never, { status?: CourseStatus }>,
+  next: NextFunction,
+) => {
+  const { id: courseId } = req.params;
+  const { action: assessmentAction } = req.body;
+
+  try {
+    const { status } = await getStatusProvider(courseId);
+
+    if (status !== CourseStatus.assessment) {
+      throw new BadRequestError(`Can not manage assessment while course in status: ${status}.`);
+    }
+
+    let statusToSet: CourseStatus;
+
+    switch (assessmentAction) {
```

Comment:
could it be posible to replace switch/case to some functions instead of

Reply:
I thought of it. But `if` construction in my opinion will look not so clear as `switch/case`. What is more, we can have additional action type in the future, for example `retrain`. 
Please, respond me if you're not agree still and let's think over changes that could be applied instead of this code.